strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiMetadataParser.java

```diff
@@ -0,0 +1,85 @@
+package org.carlspring.strongbox.util;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.carlspring.strongbox.domain.PypiPackageInfo;
+
+import org.carlspring.strongbox.util.annotations.PypiMetadataKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.*;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.TreeMap;
+
+@Component
+public class PypiMetadataParser
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(PypiMetadataParser.class);
+
+    
+    public static PypiPackageInfo parseMetadataFile(FileInputStream fis)
```

Comment:
Thanks for changing this, but would you mind changing this to use `InputStream` instead?

Reply:
Sure. Done. Kindly check 
Thanks!